myriad-scheduler/src/main/java/com/ebay/myriad/MyriadModule.java

```diff
@@ -95,11 +93,21 @@ protected void configure() {
 
     @Provides
     @Singleton
-    SchedulerState providesSchedulerState(MyriadConfiguration cfg,
-        State stateStore) {
-
+    SchedulerState providesSchedulerState(MyriadConfiguration cfg) {
         LOGGER.debug("Configuring SchedulerState provider");
-        MyriadState state = new MyriadState(stateStore);
-        return new SchedulerState(state);
+        MyriadStateStore myriadStateStore = null;
+        if (cfg.isHAEnabled()) {
+            myriadStateStore = providesMyriadStateStore();
```

Comment:
if HA is enabled and myriadStateStore is null, we should throw a RunTimeException that'd bring down RM. The message in the exception should state that  ```yarn.resourcemanager.store.class```  should be set to ```MyriadFileSystemRMStateStore``` in yarn-site.xml.